Builds fail due to implicit cmake4 upgrade

<!-- For information about contributing see: https://github.com/supercollider/supercollider/wiki/Contributing-directory -->
<!-- Please check there isn't already an issue on this topic. -->

## Environment

* SuperCollider version: dev
* Operating system: linux/windows
* Other details (Qt version, audio driver, etc.): see CI

## Steps to reproduce

Currently the CI fails due to cmake compatibility, see https://github.com/supercollider/supercollider/actions/runs/14215760688/job/39835816096#step:8:49

```shell
CMake Error at external_libraries/hidapi/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```


I don't know why this happens though b/c ubuntu 22.04 should still use [`3.22.1-1ubuntu1`](https://packages.ubuntu.com/jammy/cmake)  and 24.04 should use [`3.28.3-1build7`](https://packages.ubuntu.com/noble/cmake) which we can verify in the CI - see https://github.com/supercollider/supercollider/actions/runs/14215760688/job/39835816096#step:5:230 - so I don't understand why we are affected by a CMake 4.0 removal, see https://cmake.org/cmake/help/latest/release/4.0.html#deprecated-and-removed-features

> Compatibility with versions of CMake older than 3.5 has been removed. Calls to [cmake_minimum_required()](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html#command:cmake_minimum_required) or [cmake_policy()](https://cmake.org/cmake/help/latest/command/cmake_policy.html#command:cmake_policy) that set the policy version to an older value now issue an error. Note that calls to those commands can still support older versions of CMake by using their VERSION arguments' <min>...<max> syntax. This requires only the <min> version of CMake, but when running a newer version, sets policies up to the <max> version.

I don't really understand why the CI runs into problems here, but it is best to simply update

https://github.com/supercollider/hidapi/blob/976cb116102767dd72d58c191dd79236fca238d7/CMakeLists.txt#L1

to 3.5 and update the dependency upstream in the SC repo

edit: this also affects portaudio, see https://github.com/PortAudio/portaudio/issues/1015

--- a/external_libraries/portaudio/CMakeLists.txt
+++ b/external_libraries/portaudio/CMakeLists.txt
@@ -13,6 +13,10 @@ set(PA_LIBNAME_ADD_SUFFIX OFF CACHE BOOL "PortAudio static library suffix" FORCE
 # disable install target
 set(PA_DISABLE_INSTALL ON CACHE BOOL "Disable PortAudio install" FORCE)
 
+# bump up cmake version to fix compatibility with cmake 4
+# FIXME: remove this after the portaudio submodule is updated above v19.7.0
+set(CMAKE_POLICY_VERSION_MINIMUM 3.10)
+
 add_subdirectory(portaudio_submodule)
 
 if(WIN32 AND NOT PA_USE_ASIO)

